object-api-multipart.go

```diff
@@ -347,11 +380,22 @@ func (o objectAPI) PutObjectPart(bucket, object, uploadID string, partID int, si
 	if size > 0 {
 		if _, e = io.CopyN(multiWriter, data, size); e != nil {
 			safeCloseAndRemove(fileWriter)
-			if e == io.ErrUnexpectedEOF {
+			if e == io.ErrUnexpectedEOF || e == io.ErrShortWrite {
 				return "", probe.NewError(IncompleteBody{})
 			}
 			return "", probe.NewError(e)
 		}
+		// Allocating a byte to read from the reader.
+		// This validate whether more data exists even after 'size' number of bytes.
+		singleByte := make([]byte, 1)
```

Comment:
To read a single byte you don't need to do this. You can just do, avoids allocating a byte 

```
if _, e = io.CopyN(ioutil.Discard, data, 1); e == nil {
```